tests/test_toa_indexing.py

```diff
@@ -85,7 +77,6 @@ def test_getitem_slice(c):
     s = toas[c]
     assert set(s.get_mjds()) == set(m[c])
     if len(s) > 0:
-        assert np.all(s.table["mjd_float"] == s.table.group_by("obs")["mjd_float"])
```

Comment:
This test can be greatly simplified now that we preserve order - you can just check that `toas[c].get_mjds() == toas.get_mjds()[c]` and drop the whole rest of the testing code. Other indexing tests can be similarly modified to be both simpler and stricter.